server/scripts/import-tests/index.js

```diff
@@ -159,277 +151,206 @@ const ariaat = {
      */
-    async upsertAPGExample(
+    async upsertTestPlan(
         exampleDir,
         exampleName,
-        testVersionID,
+        ariaAtRepo,
+        commitHash,
+        commitMsg,
+        commitDate,
         exampleRefLine,
         practiceGuidelinesRefLine
     ) {
-        const exampleResult = await client.query(
-            'SELECT id, example, design_pattern FROM apg_example WHERE directory=$1 and test_version_id=$2',
-            [exampleDir, testVersionID]
-        );
-
-        let example = exampleResult.rowCount
-            ? exampleResult.rows[0]
-            : undefined;
-
-        if (!example) {
-            example = await db.ApgExample.create({
-                directory: exampleDir,
-                name: exampleName,
-                test_version_id: testVersionID
-            });
-        }
-
-        if (exampleRefLine.length > 0 && !example.example) {
-            const exampleLink = exampleRefLine[0].split(',')[1];
-            if (validUrl.isUri(exampleLink)) {
-                await db.ApgExample.update(
-                    {
-                        example: exampleLink
-                    },
-                    {
-                        where: {
-                            id: example.id
-                        }
-                    }
-                );
-            } else {
-                console.error(
-                    `WARNING: The example link is not valid for ${exampleName}. Not writing to database.`
-                );
-            }
-        }
-
-        if (practiceGuidelinesRefLine.length > 0 && !example.design_pattern) {
-            const practiceGuidelinesLink = practiceGuidelinesRefLine[0].split(
-                ','
-            )[1];
-
-            if (validUrl.isUri(practiceGuidelinesLink)) {
-                await db.ApgExample.update(
-                    {
-                        design_pattern: practiceGuidelinesLink
-                    },
-                    {
-                        where: {
-                            id: example.id
-                        }
-                    }
-                );
-            } else {
-                console.error(
-                    `WARNING: The design pattern link is not valid for ${exampleName}. Not writing to database.`
-                );
+        const getReferenceUrl = referenceLine => {
+            let url = null;
+            if (referenceLine.length) {
+                const [referenceType, link] = referenceLine[0].split(',');
+                if (validUrl.isUri(link)) url = link;
+                else
+                    console.error(
+                        `WARNING: The ${referenceType} link ${link} is not valid for ${exampleName}. Not writing to database.`
+                    );
             }
-        }
+            return url;
+        };
+
+        const exampleUrl = getReferenceUrl(exampleRefLine);
+        const designPattern = getReferenceUrl(practiceGuidelinesRefLine);
```

Comment:
I've sort of imagined exampleUrl as replacing the design pattern. Is it possible to eliminate the design pattern here?

Reply:
So the `exampleUrl` and the `APG designPatternUrl` are different properties [(example 1)](https://github.com/w3c/aria-at/blob/master/tests/checkbox/data/references.csv) and [(example 2)](https://github.com/w3c/aria-at/blob/master/tests/modal-dialog/data/references.csv), so I'm not entirely sure we want to with that stance. But if there is a reason to no longer have one, that could be discussed.
However, given that the `designPattern` lives inside the json object and isn't tied to anything, we could also ignore it for now should it become useful.

Reply:
Well, I do know we're going to talk about [the references.csv](https://github.com/w3c/aria-at/issues/345) pretty soon as one of the first topics of the larger test authoring format discussion. So yes, I think ... we should remove it.

Reply:
I don't think we need to get rid of it unless we don't use it at all. The design pattern is a term that relates to the types of components in the example. See this issue as a reference: https://github.com/w3c/aria-at/issues/402. I think it makes more sense to keep it as it is something that is heavily used in APG. As long as nothing is dependent upon it we can use or not use it as the examples indicate.

Reply:
I agree with @jesdaigle that we should not get rid of `designPatternUrl`. 

There are multiple examples for each APG design pattern. For example, "Checkbox Example (Two State)" is one of two examples under the "Checkbox" design pattern. Each URL has different, but useful contextual information. The CG wants to make it easy to visit either of these links for context for reviewers and testers.

For now, I think we should just track `references.csv`.

In the future, I agree with @alflennik that we should advocate for APG-agnostic metadata. This could be accomplished by a rename like: `APG Pattern URL → Context URL` or something like that, or it could be accomplished by a freeform json object that has different properties for each source that we're testing (APG, HTML, etc.). We should definitely discuss the pros/cons of these down the road.

Reply:
Thanks for that context @jesdaigle and @s3ththompson I didn't realize the test plans had a hierarchy like that!